src/hotspot/cpu/x86/x86_64.ad

```diff
@@ -4500,11 +4500,23 @@ instruct maxF_reg(legRegF dst, legRegF a, legRegF b, legRegF tmp, legRegF atmp,
   %}
   ins_encode %{
     int vector_len = Assembler::AVX_128bit;
-    __ vblendvps($btmp$$XMMRegister, $b$$XMMRegister, $a$$XMMRegister, $b$$XMMRegister, vector_len);
-    __ vblendvps($atmp$$XMMRegister, $a$$XMMRegister, $b$$XMMRegister, $b$$XMMRegister, vector_len);
-    __ vmaxss($tmp$$XMMRegister, $atmp$$XMMRegister, $btmp$$XMMRegister);
-    __ vcmpps($btmp$$XMMRegister, $atmp$$XMMRegister, $atmp$$XMMRegister, Assembler::_false, vector_len);
-    __ vblendvps($dst$$XMMRegister, $tmp$$XMMRegister, $atmp$$XMMRegister, $btmp$$XMMRegister, vector_len);
+    XMMRegister mask = $b$$XMMRegister;
+    if (EnableX86ECoreOpts) {
+      __ vpsrad($tmp$$XMMRegister, mask, 32, vector_len);
+      mask = $tmp$$XMMRegister;
+    }
+    __ vblendvps($btmp$$XMMRegister, $b$$XMMRegister, $a$$XMMRegister, mask, vector_len, true, $atmp$$XMMRegister);
+    __ vblendvps($atmp$$XMMRegister, $a$$XMMRegister, $b$$XMMRegister, mask, vector_len, true, $tmp$$XMMRegister);
+
+    if ($dst$$XMMRegister == $btmp$$XMMRegister) {
+      __ vmaxss($btmp$$XMMRegister, $atmp$$XMMRegister, $btmp$$XMMRegister);
+      __ vcmpps($tmp$$XMMRegister, $atmp$$XMMRegister, $atmp$$XMMRegister, Assembler::_false, vector_len);
+      __ vblendvps($dst$$XMMRegister, $btmp$$XMMRegister, $atmp$$XMMRegister, $tmp$$XMMRegister, vector_len, true, $tmp$$XMMRegister);
+    } else {
+      __ vmaxss($tmp$$XMMRegister, $atmp$$XMMRegister, $btmp$$XMMRegister);
+      __ vcmpps($btmp$$XMMRegister, $atmp$$XMMRegister, $atmp$$XMMRegister, Assembler::_false, vector_len);
+      __ vblendvps($dst$$XMMRegister, $tmp$$XMMRegister, $atmp$$XMMRegister, $btmp$$XMMRegister, vector_len, true, $btmp$$XMMRegister);
+    }
```

Comment:
Please move into a new macro assembly routine.

Reply:
moved to existing routine